Fix errors, grammar, and consistency in core format documentation (#576)
apache/parquet-format: Encodings.md

For native types, this outputs the data as little endian. Floating
    point types are encoded in IEEE.

For the byte array type, it encodes the length as a 4-byte little

> endian, followed by the bytes.

Reviewer: ```suggestion
endian integer, followed by the bytes.
```

Author: Done.

Dictionary page format: the entries in the dictionary using the [plain](#PLAIN) encoding.

Data page format: the bit width used to encode the entry ids stored as 1 byte (max bit width = 32),

> followed by the values encoded using RLE/Bit-Packed described above (with the given bit width).

Reviewer: ```suggestion
followed by the values encoded using the RLE/Bit-Packing described above (with the given bit width).
```

Author: Done.

* Dictionary indices
* Boolean values in data pages, as an alternative to PLAIN encoding

> Whether prepending the four-byte `length` to the `encoded-data` is summarized in the table below:

Reviewer: ```suggestion
Whether or not to prepend the four-byte `length` to the `encoded-data` is summarized in the table below:
```

Author: Done.

Supported Types: BYTE_ARRAY

This encoding is always preferred over PLAIN for byte array columns.

> 

Reviewer: This isn't a typo, you're actually changing the spec here. I think this needs an actual discussion.

Author: Reverted. You are right, this is a spec change and should be discussed separately.
